Sources/DequeModule/Deque.swift

```diff
@@ -101,7 +130,102 @@ public struct Deque<Element> {
+  ///
+  /// Fixed-capacity deques act as ring buffers, automatically removing the
+  /// oldest elements when new elements are added to a full deque. This makes
+  /// them ideal for scenarios requiring predictable memory usage, such as
+  /// audio processing, logging systems, or real-time applications.
+  ///
+  /// - Parameter fixedCapacity: The maximum number of elements that the
+  ///   deque can hold. Must be greater than zero.
+  @inlinable
+    public init(fixedCapacity: Int){
+        precondition(fixedCapacity > 0, "Fixed capacity must be greater than zero")
+        self._storage = _Storage(minimumCapacity: fixedCapacity)
+        self._maxCapacity = fixedCapacity
+    }
+}
+
+extension Deque {
+    
+    /// A Boolean value that indicates whether this deque was created with a fixed
+    /// maximum capacity.
+    ///
+    /// - Returns: `true` if the deque is a fixed-capacity deque (ring buffer),
+    ///   `false` if it can grow without limit.
+    ///
+    /// ### Example
+    ///
+    ///     var regular = Deque<Int>()
+    ///     print(regular.isFixedCapacity) // false
+    ///
+    ///     var buffer = Deque<Int>(fixedCapacity: 5)
+    ///     print(buffer.isFixedCapacity) // true
+    ///
+    /// Use this property when you need to distinguish between a
+    /// regular deque and a fixed-capacity deque.
+    ///
+    /// - Complexity: O(1)
+    @inlinable
+    public var isFixedCapacity: Bool {
+        return _maxCapacity != nil
+    }
+    
+    /// The maximum number of elements this deque can hold, or `nil` if it
+    /// has no fixed limit.
+    ///
+    /// Example:
+    ///
+    ///     var buffer = Deque<Int>(fixedCapacity: 3)
+    ///     print(buffer.maxCapacity) // Optional(3)
+    ///
+    ///     var regular = Deque<Int>()
+    ///     print(regular.maxCapacity) // nil
+    ///
+    /// - Complexity: O(1)
+    @inlinable
+    public var maxCapacity: Int? {
+        return _maxCapacity
+    }
+    
+    /// Returns true if the deque is at its maximum capacity.
+    ///
+    /// Example:
+    ///
+    ///     var buffer = Deque<Int>(fixedCapacity: 2)
+    ///     buffer.append(1)
+    ///     buffer.append(2)
+    ///     print(buffer.isFull) // true
+    ///
+    /// When the deque is full, adding new elements will remove the oldest elements.
+    @inlinable
+    public var isFull: Bool {
+        guard let maxCap = _maxCapacity else { return false }
+        return count >= maxCap
+    }
+
+    /// The number of additional elements that can be inserted before the deque
+    /// becomes full.
+    ///
+    /// For regular (unlimited) deques, this value is very large (`Int.max`).
+    ///
+    /// Example:
+    ///
+    ///     var buffer = Deque<Int>(fixedCapacity: 3)
+    ///     print(buffer.remainingCapacity) // 3
+    ///     buffer.append(1)
+    ///     print(buffer.remainingCapacity) // 2
+    ///
+    /// - Complexity: O(1)
+    @inlinable
+    public var remainingCapacity: Int {
```

Comment:
For `RigidArray` and `RigidDeque` (and later `RigidSet`, `RigidDictionary` etc) I've been using the name `freeCapacity` for the same property.

It feels like your name may be a better choice! I'll sleep on it -- we may want `RigidArray` to adopt this before we ship 1.3.0.

Reply:
D'oh, we have already established the `freeCapacity` name in the `OutputSpan` type that has shipped in Swift 6.2's stdlib. 

Oh well; `remainingCapacity` feels better, but it's better to be consistent than perfect.